compiler/testData/diagnostics/tests/delegatedProperty/inference/differentDelegatedExpressions.fir.kt

```diff
@@ -12,8 +12,8 @@ class A(outer: Outer) {
 
     var b: String by  foo(getMyProperty())
     var r: String by  foo(outer.getContainer().getMyProperty())
-    var e: String by  <!DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE!>+ foo(getMyProperty())<!>
-    var f: String by  <!DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE!>foo(getMyProperty()) - 1<!>
+    var e: String by  <!DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE, DELEGATE_SPECIAL_FUNCTION_RETURN_TYPE_MISMATCH!>+ foo(getMyProperty())<!>
+    var f: String by  <!DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE, DELEGATE_SPECIAL_FUNCTION_RETURN_TYPE_MISMATCH!>foo(getMyProperty()) - 1<!>
```

Comment:
Hm, actually, neither is reported?

https://github.com/JetBrains/kotlin/blob/ad9b962536e0d0c148baf884111c2c97c2bc8f2c/compiler/testData/diagnostics/tests/delegatedProperty/inference/differentDelegatedExpressions.kt#L15-L16

Reply:
Aha! Thanks! So it looks FE1.0 suppress reporting in this case when the inference for type parameters failed. Let me try mimic this behavior.

Reply:
So we need to report NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER and suppress delegate issues in that case. But I just saw in https://docs.google.com/document/d/1f_uhx9LUqFD0Kl6V70rw03GFkBH3nVoGoA5RjcekNrM/edit#heading=h.j1zcgdakihi4 that says JetBrains folks will take care of it. Let me add a TODO here then.